code/game/objects/items/miscellaneous.dm

```diff
@@ -101,6 +101,51 @@
 	new /obj/item/toy/crayon/spraycan(src)
 	new /obj/item/clothing/shoes/sandal(src)
 
+/obj/item/choice_beacon/augments
+	name = "augment beacon"
+	desc = "Summons augmentations. Can be used 3 times!"
+	var/uses = 3
+
+/obj/item/choice_beacon/augments/generate_display_names()
+	var/static/list/augment_list
+	if(!augment_list)
+		augment_list = list()
+		var/list/templist = list(
+		/obj/item/organ/cyberimp/brain/anti_drop,
+		/obj/item/organ/cyberimp/arm/toolset,
+		/obj/item/organ/cyberimp/arm/surgery,
+		/obj/item/organ/cyberimp/chest/thrusters,
+		/obj/item/organ/lungs/cybernetic/upgraded,
+		/obj/item/organ/liver/cybernetic/upgraded) //cyberimplants range from a nice bonus to fucking broken bullshit so no subtypesof
+		for(var/V in templist)
+			var/atom/A = V
+			augment_list[initial(A.name)] = A
+	return augment_list
+
+/obj/item/choice_beacon/augments/generate_options(mob/living/M)
+	var/list/display_names = generate_display_names()
+	if(!display_names.len)
+		return
+	var/choice = input(M,"Which item would you like to order?","Select an Item") as null|anything in display_names
+	if(!choice || !M.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
+		return
+
+	spawn_option(display_names[choice],M)
+	uses--
```

Comment:
You decrease uses after spawning and before input, this is exploitable.

Reply:
the input is on line 129, the spawn option is on line 133, and we decrease the uses on line 134. Am I missing something?

Reply:
Input is blocking. Open 99 inputs, then click yes 99 times to get 99 items.